src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -3455,6 +3487,110 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
         }
     }
 }
+//----------------------------------------------------------------------------------------------
+// Lowering::LowerHWIntrinsicCndSel: Lowers a Sve ConditionalSelect call
+//
+//  Arguments:
+//     node - The hardware intrinsic node of the form
+//            ConditionalSelect(mask, trueValue, falseValue)
+//
+GenTree* Lowering::LowerHWIntrinsicCndSel(GenTreeHWIntrinsic* cndSelNode)
+{
+    assert(cndSelNode->OperIsHWIntrinsic(NI_Sve_ConditionalSelect));
+
+    GenTree* op1 = cndSelNode->Op(1);
+    GenTree* op2 = cndSelNode->Op(2);
+    GenTree* op3 = cndSelNode->Op(3);
+
+    if (op1->OperIsHWIntrinsic())
+    {
+        GenTreeHWIntrinsic* predNode = op1->AsHWIntrinsic();
+        if (predNode->IsAllFalse())
+        {
+            // Transform:
+            //
+            // CndSel(AllFalse, op2, op3) to
+            // op3
+            op1->SetUnusedValue();
+            op2->SetUnusedValue();
+            cndSelNode->SetUnusedValue();
+
+            return op3;
+        }
```

Comment:
Is this needed? Shouldn't it be getting handled by the general morph optimizations instead, particularly since we shouldn't ourselves be introducing `CndSel(false, x, y)` in lowering?

Reply:
I think this was one before your changes in https://github.com/dotnet/runtime/pull/104175. I will have to recheck if this is needed now.

Reply:
yep, not needed anymore.